server/routes/admin.js

```diff
@@ -0,0 +1,29 @@
+import { Router } from 'express'
+
+import { collections } from '../utils/mongoCollections'
+import ensureAdmin from '../utils/passport/ensure-admin'
+
+const router = Router()
+
+router.route('/api/v1/admin/:uid').patch(ensureAdmin, async (req, res) => {
+  try {
+    const { appDb } = req.app.locals
+    const { uid } = req.params
+    const updatedUserAccess = await appDb
+      .collection(collections.users)
+      .updateOne(
+        { uid },
+        {
+          $set: req.body,
+        }
+      )
+
+    return updatedUserAccess.modifiedCount === 1
+      ? res.status(200).json({ data: updatedUserAccess })
+      : res.status(404).json({ message: 'User could not be updated' })
+  } catch (error) {
+    return res.status(404).json({ message: 'User could not be updated' })
+  }
+})
```

Comment:
I think we should continue the pattern of having a controller handle requests, so this would be

```suggestion
router.route('/api/v1/admin/:uid').patch(ensureAdmin, adminUsersController.update)
```

and then we need to create an `adminUsersController`

```js
const AdminUsersController = {
  update: async (req, res) => {
    try {
      const { appDb } = req.app.locals
      const { uid } = req.params
      const updatedUser = await appDb
        .collection(collections.users)
        .updateOne(
          { uid },
          { $set: req.body },
        )

      return updatedUser.modifiedCount === 1
        ? res.status(200).json({ data: updatedUserAccess })
        : res.status(404).json({ message: 'User could not be updated' })
    } catch (error) {
      return res.status(404).json({ message: 'User could not be updated' })
    }
  },
```

This should make it easier to test the functionality in the admin users controller as well.

Reply:
Thanks for this. It looks different than the one in the refactor, so I'm guessing this is how the api charts routes are going to be..?

Reply:
I believe this is a similar/same pattern as the refactor. What do you see as being different?

Reply:
I think we want to be consistent with this, seeing as how there are a lot of inconsistencies throughout the project. Personally I like the wrapper, but maybe this is a discussion for the modernization 🤷 

The refactor has it as
```javascript
const index = indexStuff
const show = showStuff

export default {
index, show
}
```

Here we are naming the controller and using an object for our methods. 

```javascript
const SomeController = {
  update: updateStuff
}

export default SomeController
```

Reply:
oh yeah either style is fine, I think the gist is we want the route handling to be in a controller.